### PR TITLE
[WIP] Convert e2e tests to a Job

### PR DESCRIPTION
**Pull request by ironcladlou**
Fix the e2e tests and convert them to a k8s Job.

### CODE REVIEW

**Comment by ironcladlou** [comment]
Not sure if this is totally the right way to go about things, but it's something to talk about.

File: test/e2e/e2e-job-template.yaml [hunk 1]
```diff
@@ -0,0 +1,27 @@
+apiVersion: template.openshift.io/v1
```
Comment (brancz) [inline test/e2e/e2e-job-template.yaml]: As our goal with this project is to eventually be compatible with vanilla I would prefer not to introduce any openshift specific means if not absolutely necessary.

File: test/e2e/main_test.go [hunk 2]
```diff
@@ -67,53 +62,72 @@ func TestQueryPrometheus(t *testing.T) {
...
 	}
 
-	err = wait.Poll(5*time.Second, 10*time.Minute, func() (bool, error) {
+	err = wait.Poll(5*time.Second, 20*time.Second, func() (bool, error) {
 		defer t.Log("---------------------------\n")
 
 		for _, q := range queries {
-			n, err := f.QueryPrometheus("prometheus-k8s-0", q.query)
+			response, err := client.query(q.query)
+			res, err := gabs.ParseJSONBuffer(response.Body)
```
Comment (brancz) [inline test/e2e/main_test.go]: We actually do know the format of this response so we should actually parse it correctly and not use this dependency.

File: test/e2e/e2e-job-template.yaml (hunk 1 above)
Comment (ironcladlou) [reply test/e2e/e2e-job-template.yaml]: Good point- also, @smarterclayton later yesterday evening pointed me to docs for the new ci-operator process which I think would provide the path forward here. I'll probably redo this PR along those lines.

File: test/e2e/main_test.go (hunk 2 above)
Comment (ironcladlou) [reply test/e2e/main_test.go]: I wasn't entirely sure why we weren't unmarshaling into a little struct, but I was just doing a quick refactor. Will take care of this too.

**Pull request closed without merging**